tuf/ngclient/fetcher.py

```diff
@@ -72,21 +73,21 @@ def fetch(self, url: str) -> Iterator[bytes]:
 
     @contextmanager
     def download_file(self, url: str, max_length: int) -> Iterator[IO]:
-        """Opens a connection to 'url' and downloads the content
-        up to 'max_length'.
+        """Opens a connection to ``url`` and downloads the content
+        up to ``max_length``.
```

Comment:
Minor nit: According to our style guide, this should fit on one line. Also, at first glance this could be read as if hitting max length would just stop the download, although it actually raises an exception, which is described below anyway. Maybe we can just say as we do in `download_bytes`?
```suggestion
        """Download file from given url.
```
?

Reply:
Optional: If we do want to give any more info here, e.g. in a paragraph after the opening line, I'd probably mention that `download_file` should be used in a `with` block, to always release the allocated file resources even if download fails. (not a blocker for this PR).

Reply:
Minor nit 2: Consider consistently wrapping `url` in double backticks in opening lines of all functions of this module